App.js

```diff
@@ -57,13 +80,126 @@ const TabNavigator = createBottomTabNavigator({
     })
   });
 
+  const showAlert = ((title, body) => {
+    Alert.alert(
+      title, body
+    );
+  })
+
 const AppContainer = createAppContainer(TabNavigator);
 
 export default class App extends Component<Props> {
 
+  constructor(props) {
+    super(props)
+    firebase.messaging().subscribeToTopic("all");
+    console.log("Subscribed to topic");
+    console.log("Checking Status")
+    this.checkStatus();
+    let curr = new Date();
+
+    Object.keys(events).forEach((event) => {
+      if (events[event.toString()].date == (curr.getDate())) {
+        if (events[event].timeStart - (curr.getTime()) <= 15 * 60 * 1000) {
+          showAlert(event, "Starting soon!");
+        }
+      }
+    })
+  }
+
+  async checkStatus() {
+    BackgroundFetch.status((status) => {
+      switch(status) {
+        case BackgroundFetch.STATUS_RESTRICTED:
+          console.log("BackgroundFetch restricted");
+          break;
+        case BackgroundFetch.STATUS_DENIED:
+          console.log("BackgroundFetch denied");
+          break;
+        case BackgroundFetch.STATUS_AVAILABLE:
+          console.log("BackgroundFetch is enabled");
+          break;
+      }
+    })
+  }
+
+  localNotifs() {
+    console.log("Checking schedule");
+    BackgroundFetch.configure({
+      minimumFetchInterval: 15,     // <-- minutes (15 is minimum allowed)
+      // Android options
+      stopOnTerminate: false,
+      startOnBoot: true,
+      requiredNetworkType: BackgroundFetch.NETWORK_TYPE_NONE, // Default
+      requiresCharging: false,      // Default
+      requiresDeviceIdle: false,    // Default
+      requiresBatteryNotLow: false, // Default
+      requiresStorageNotLow: false  // Default
+    }, () => {
+      console.log("[js] Received background-fetch event");
+      // Required: Signal completion of your task to native code
+      // If you fail to do this, the OS can terminate your app
+      // or assign battery-blame for consuming too much background-time
+      let curr = new Date();
+      console.log("checking background events");
+      Object.keys(events).forEach((event) => {
+        if (events[event.toString()].date == (curr.getDate())) {
```

Comment:
this check actually shouldn't be necessary with unix timestamp, should also make midnight edge case irrelevant